### DISCUSSION TITLE
Runtime packages - finalize

### DISCUSSION THREAD

**Discussion by freddydk** [microsoft/AL-Go#1606] [Ideas]
Now, we have symbols from AppSource Apps and Microsoft Apps - we need to go the next step.

- [ ] How does partners generate and publish runtime packages to NuGet easily
- [ ] Document how to share NuGet packages with other partners
- [ ] Document how to trust partner NuGet packages (using PATs, OAuth and federated credentials)

**Comment by ghost** [comment]
That is, status today, exactly one of the features that is still stopping us from switching. 

Of course, the ideal solution would be to create the runtime packages without the presence of a container. 🫠
I know...Another story...Not really in your own hands...

So, well...
At this point I would like to say a big thank you for your work. Simply great.♥️

**Comment by jonaswre** [comment]
We have a flow that’s working, more or less.

I’ve updated the repo that @freddydk created for Runtime generation. I reworked it so the Runtime generation can now be triggered just by specifying the repo name.

The action grabs the latest release from the specified repo and resolves all dependencies using the internal NuGet feed. Once the runtimes are generated, the resulting NuGet package is uploaded to a second organization. From there, we invite our partner and grant them access to the runtimes.

This setup has a few drawbacks. First, everyone you invite can see each other, which isn’t ideal. Second, the UI is clunky—you have to use PowerShell, and there’s no GUI. Third, it’s horribly inefficient. We’re building some of our apps from versions 21.0 to 25.1 in every language for every release. That’s a ton of wasted compute cycles. But as long as Microsoft is footing the bill, I’m not too concerned. If it ever gets too expensive for them, they can deal with fixing the runtime situation themselves.

Another quirk: you need a separate BC Runtime repo for every App repo. If you don’t, builds from different apps might cancel each other out. That said, since Microsoft is paying, the best strategy might be to create a new organization for every App repo. This would let you parallelize runtime creation even further, though I’m not sure how many organizations you can actually create before hitting a limit.

Honestly, it’s a waste of time trying to work with this runtime mess. Microsoft needs to stop chasing AI hype and focus on building a proper framework—one that allows partners to spend their time delivering value to customers instead of dealing with these problems.

**Comment by jonaswre** [comment]
![Image](https://github.com/user-attachments/assets/f343a835-dee2-4f7a-93d4-783d99e4b965)
Thats 1/3 of our included runner minutes if we had to pay it. Just to get a few runtimes.

**Comment by louagej** [comment]
Hi there,

Is this feature arriving soon, or is it just a mirage on the distant horizon?

**Comment by freddydk** [comment]
Cannot give you a date, but it is very close to the top of my backlog (there are other things in the AL-Go backlog - but this is on the list of stuff, that I need to throw my love on)

**Comment by PropmanProphecy** [comment]
what is the status on the issue since Freddy no longer is at Microsoft

**Comment by gntpet** [comment]
Are there any plans to release this feature?

**Comment by fvet** [comment]
@aholstrup1 @mazhelez Any status update on this feature?

**Reply by mazhelez** [reply]
Hi @fvet 

It is being worked on. The direction has shifted a bit, but I hope we'll be able to share more details soon!

**Reply by frottke** [reply]
hi @mazhelez,

might this direction be related to the new parameter “CreateReadyToRun” in the PowerShell command “Publish-NavApp”? 

I stumbled across the new parameter a few weeks ago and couldn't find any explaining documentation regarding the parameter. Only hint I found was in the dll which had a procedure for creating an app manifest for ReadyToRun apps.

**Reply by PropmanProphecy** [reply]
@aholstrup1  @mazhelez  is there any news when this is ready?

**Reply by mazhelez** [reply]
Unfortunately, no. We hope to be able to share more details soon.

**Reply by rub7** [reply]
Bumping this. Any updates? This feature is what is preventing us from migrating to AL-Go
